sktime/forecasting/compose/_networkpipeline.py

```diff
@@ -0,0 +1,278 @@
+            # estimators have fit and predict methods
+            if hasattr(est, "fit") and hasattr(est, "predict"):
+                f = clone(est)
+
+                f.fit(**processed_arguments)
+                self._fitted_estimators[name] = f
+
+        self._is_fitted = True
+
+        return self
+
+    def _predict(self, fh=None, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        if X is not None:
+            self._X = X
+        for name, est, arguments in self._iter():
+            processed_arguments = {}
+            # estimator in self.steps is not fitted.
+            # use fitted estimator in self._fitted_estimators
+            est = self._fitted_estimators[name]
+            if "predict" in arguments and arguments["predict"] is None:
+                # this step must be skipped
+                continue
+            elif "predict" in arguments:
+                processed_arguments = self._process_arguments(arguments["predict"])
+            else:
+                processed_arguments = self._process_arguments(arguments)
+            if hasattr(est, "transform"):
+
+                self._step_results[name] = est.transform(**processed_arguments)
+            if hasattr(est, "predict"):
+
+                pred = est.predict(
+                    **processed_arguments, return_pred_int=return_pred_int, alpha=alpha
+                )
+                # pred.index.freq = self._y.index.freq
+                self._step_results[name] = pred
+
+        return self._step_results[self.steps[-1][0]]
+
+    def update(self, y, X=None, update_params=True):
+        """Update fitted parameters
+
+        Parameters
+        ----------
+        y : pd.Series
+        X : pd.DataFrame
+        update_params : bool, optional (default=True)
+
+        Returns
+        -------
+        self : an instance of self
+        """
+        self.check_is_fitted()
+        y_index_frequency = self._y.index.freq
+        self._update_y_X(y, X)
+        self._y = self._y.asfreq(y_index_frequency)
+        for name, est, arguments in self._iter():
+            # estimator in self.steps is not fitted.
+            # use fitted estimator in self._fitted_estimators
+            est = self._fitted_estimators[name]
+            if "update" in arguments and arguments["update"] is None:
+                # this step must be skipped
+                continue
+            # assume update takes same arguments as fit
+            elif "update" in arguments:
+
+                processed_arguments = self._process_arguments(arguments["update"])
+
+            else:
+                processed_arguments = self._process_arguments(arguments)
+            # Transformers are instances of BaseTransformer and BaseEstimator
+            # Estimators are only instances of BaseEstimator
+
+            if hasattr(est, "update"):
+                out = est.update(**processed_arguments)
+                self._step_results[name] = out
+                self._is_fitted = True
```

Comment:
We shouldn't update the `_is_fitted` state anywhere but in `fit(...)`.

Reply:
ok